tests/unit-tests/test-class-sensei-analysis-overview-list-table.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * Tests for Sensei_Analysis_Overview_List_Table class.
+ */
+class Sensei_Analysis_Overview_List_Table_Test extends WP_UnitTestCase {
+
+	/**
+	 * Set up before each test.
+	 */
+	public function setup() {
+		parent::setUp();
+
+		$this->factory = new Sensei_Factory();
+	}
+
+	/**
+	 * Tear down after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		$this->factory->tearDown();
+	}
+
+	/**
+	 * Tests the last activity logic.
+	 *
+	 * @covers Sensei_Admin::get_last_activity
+	 */
+	public function testGetLastActivity() {
+		/* Arrange. */
+		$user_id  = $this->factory->user->create();
+		$lesson_1 = $this->factory->lesson->create(
+			[
+				'meta_input' => [
+					'_lesson_course' => $this->factory->course->create(),
+				],
+			]
+		);
+		$lesson_2 = $this->factory->lesson->create(
+			[
+				'meta_input' => [
+					'_lesson_course' => $this->factory->course->create(),
+				],
+			]
+		);
+
+		$instance = new Sensei_Analysis_Overview_List_Table();
+		$method   = new ReflectionMethod( $instance, 'get_last_activity' );
+		$method->setAccessible( true );
+
+		/* Act. */
+		// Start lesson 1 (status: in-progress).
+		$lesson_1_activity_comment_id = Sensei_Utils::sensei_start_lesson( $lesson_1, $user_id );
+
+		/* Assert. */
+		$this->assertEmpty(
+			$method->invoke( $instance, $user_id ),
+			'The last activity should not take into account activities that are in progress.'
+		);
+
+		/* Act. */
+		// Simulate quiz submission.
+		wp_update_comment(
+			[
+				'comment_ID'       => $lesson_1_activity_comment_id,
+				'comment_approved' => 'ungraded',
+			]
+		);
+
+		/* Assert. */
+		$this->assertNotEmpty(
+			$method->invoke( $instance, $user_id ),
+			'The last activity should take into account activities with status "ungraded".'
+		);
+
+		/* Act. */
+		// Simulate passing a quiz.
+		wp_update_comment(
```

Comment:
Tests with `wp_update_comment` look like good candidates for a separate test with the data provider. 
All these tests look very similar with a difference in the in/out values.

Reply:
It's funny that you mention this because I usually split tests but here decided to not do it for some reason. 😄 
But I totally agree, splitting the tests makes them more readable and easier to maintain. Updated in 03caf181f0e9a94583aded134a7ee2d2221bbc06.

Reply:
I've done a few more test updates after the mentioned commit above.